File: packages/javascript/lib/word/randGem.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Stones and metals, precious and ordinary alike.
 *
 * `randWord({ theme: 'gem' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randGem({ language: 'ko', count: 3 }); // ['금', '수정', '흑요석']
 * randGem({ language: 'en', count: 3 }); // ['Gold', 'Quartz', 'Obsidian']
 */
export function randGem(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Gem words with the language and theme each one came from.
 *
 * @example
 * randGem({ language: 'ko', output: 'detail' });
 * // [{ word: '금', language: 'ko', theme: 'gem' }]
 */
export function randGem(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randGem(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('gem', options);
}

File: packages/dart/lib/src/word/rand_gem.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Stones and metals, precious and ordinary alike.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.gem] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randGem(language: WordLanguage.ko, count: 3); // [금, 수정, 흑요석]
/// randGem(language: WordLanguage.en, count: 3); // [Gold, Quartz, Obsidian]
/// ```
List<String> randGem({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.gem,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
